ui/lib/apps/Statement/utils/useSchemaColumns.ts

```diff
@@ -0,0 +1,28 @@
+import { useEffect, useState } from 'react'
+
+import client from '@lib/client'
+import { cache } from '@lib/utils/callableCache'
+
+const statementsTableColumnsGet = client
+  .getInstance()
+  .statementsTableColumnsGet.bind(client.getInstance())
+
+export const useSchemaColumns = () => {
+  const [isLoading, setLoading] = useState(true)
+  const [schemaColumns, setSchemaColumns] = useState<string[]>([])
+
+  useEffect(() => {
+    const fetchSchemaColumns = async () => {
+      const { data } = await cache(statementsTableColumnsGet).call()
```

Comment:
Does this mean that it will only send requests for the first time? Then there may be issues since we want to support the upgrading scenario. In this scenario, the columns may be changed.

I think we might be able to just remove this cache, or add TTL to it.

BTW I would prefer to add the cache layer to the API (inside utils), instead of inside the business part, if it is really necessary. The reason is simple that there may be multiple business parts calling this API endpoint and they should share the same cache.

Reply:
Yes, I'm considering adding it to useClientRequest.

Reply:
I suspect the design of this cache may take some time to be nice. Let's separate it out in this PR :)

Reply:
And, if we are already talking about useClientRequest + cache, then.... 😏 it may be time to consider something like [SWR](https://swr.vercel.app/)!

Reply:
That's right! SWR is a good option. We do need API cache switch in some scenario, now is global configuration.

Reply:
And I think in-memory TTL is unnecessary in client side. Maybe upgrading scenario requires its own mechanism in client side? Like refresh all cache, etc. Otherwise the upgrade logic will be separated everywhere.

Reply:
I found that [ahooks useRequest](https://ahooks.js.org/zh-CN/hooks/async) is good enough.

Reply:
A common problem for these libraries is cancellation I remember.

Reply:
About what?

https://github.com/alibaba/hooks/blob/master/packages/use-request/src/useAsync.ts#L210